### PR TITLE
Include OAuth libs into images 

### PR DESCRIPTION
**Pull request by see-quick**
This PR adds OAuth libs to our images. That way, we can possibly configure OAuth inside the test-container repository. The image size increases by only 12 MB, which is negligible, in my opinion.

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
Should we create something similar to the 3rd part libraries we have in operators repo here to add various addons? Adding it JAR by JAR does not seem to be very stable way to do this.

**Comment by see-quick** [comment]
> Should we create something similar to the 3rd part libraries we have in operators repo here to add various addons? Adding it JAR by JAR does not seem to be very stable way to do this.

Sure I can do it :).

File: images/base/Dockerfile [hunk 1]
```diff
@@ -2,6 +2,23 @@ FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
...
+
+# Download OAuth libraries
+RUN echo "Including OAuth libraries in the base image..." && \
+    mkdir -p /opt/kafka/oauth-libs && \
+    OAUTH_COMMON_URL="https://repo1.maven.org/maven2/io/strimzi/kafka-oauth-common/${OAUTH_LIB_VERSION}/kafka-oauth-common-${OAUTH_LIB_VERSION}.jar" && \
+    OAUTH_SERVER_URL="https://repo1.maven.org/maven2/io/strimzi/kafka-oauth-server/${OAUTH_LIB_VERSION}/kafka-oauth-server-${OAUTH_LIB_VERSION}.jar" && \
+    OAUTH_SERVER_PLAIN_URL="https://repo1.maven.org/maven2/io/strimzi/kafka-oauth-server-plain/${OAUTH_LIB_VERSION}/kafka-oauth-server-plain-${OAUTH_LIB_VERSION}.jar" && \
+    OAUTH_KEYCLOAK_AUTHORIZER_URL="https://repo1.maven.org/maven2/io/strimzi/kafka-oauth-keycloak-authorizer/${OAUTH_LIB_VERSION}/kafka-oauth-keycloak-authorizer-${OAUTH_LIB_VERSION}.jar" && \
+    OAUTH_CLIENT_URL="https://repo1.maven.org/maven2/io/strimzi/kafka-oauth-client/${OAUTH_LIB_VERSION}/kafka-oauth-client-${OAUTH_LIB_VERSION}.jar" && \
+    NIMBUS_JOSE_JWT_URL="https://repo1.maven.org/maven2/com/nimbusds/nimbus-jose-jwt/${NIMBUS_JOSE_JWT_VERSION}/nimbus-jose-jwt-${NIMBUS_JOSE_JWT_VERSION}.jar" && \
```
Comment (scholzj) [inline images/base/Dockerfile:22]: So if you wanna do it this way ... aren't you missing Jackson Databind and Json Path?

Comment (see-quick) [reply images/base/Dockerfile:22]: Hmm, I tried using OAuth in test containers, and it worked well without any issues. If those libs were needed, would I get an error?

Comment (scholzj) [reply images/base/Dockerfile:22]: Maybe a question on @mstruk?

Comment (scholzj) [reply images/base/Dockerfile:22]: But I guess they might be needed only in some configs?

Comment (see-quick) [reply images/base/Dockerfile:22]: > But I guess they might be needed only in some configs?
That's what I am thinking also 🤔 .

Comment (see-quick) [reply images/base/Dockerfile:22]: Okay using OAUTHBEARER method throws
```
Caused by: java.lang.ClassNotFoundException: com.fasterxml.jackson.annotation.JsonKey
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	... 53 more
```
when using producer and consumer so I think those dependencies are needed at least `Jackson Databind`

**Review by scholzj — APPROVED** [review]

**Pull request merged**